modules/dop/providers/guide/guide.service.go

```diff
@@ -0,0 +1,226 @@
+		guides = append(guides, v.Convert())
+	}
+	return &pb.ListGuideResponse{Data: guides}, nil
+}
+
+func (g *GuideService) ProcessGuide(ctx context.Context, req *pb.ProcessGuideRequest) (*pb.ProcessGuideResponse, error) {
+	if err := req.Validate(); err != nil {
+		return nil, apierrors.ErrProcessGuide.InvalidParameter(err)
+	}
+
+	if req.Kind == db.PipelineGuide.String() {
+		if err := g.db.UpdateGuideByAppIDAndBranch(req.AppID, req.Branch, req.Kind, map[string]interface{}{"status": db.ProcessedStatus}); err != nil {
+			return nil, apierrors.ErrProcessGuide.InternalError(err)
+		}
+	}
+
+	return &pb.ProcessGuideResponse{}, nil
+}
+
+func (g *GuideService) BatchUpdateGuideExpiryStatus() error {
+	return g.db.BatchUpdateGuideExpiryStatus()
+}
+
+type PipelineYml struct {
+	YmlName string
+	YmlPath string
+}
+
+func (g *GuideService) ListPipelineYml(app *apistructs.ApplicationDTO, branch, userID string) ([]PipelineYml, error) {
+	work := limit_sync_group.NewWorker(3)
+	var list []PipelineYml
+	var pathList = []string{"", DicePipelinePath, ErdaPipelinePath}
+	for _, path := range pathList {
+		work.AddFunc(func(locker *limit_sync_group.Locker, i ...interface{}) error {
+			result, err := g.getPipelineYml(app, userID, branch, i[0].(string))
+			if err != nil {
+				return nil
+			}
+
+			locker.Lock()
+			defer locker.Unlock()
+			list = append(list, result...)
+			return nil
+		}, path)
+	}
+	if err := work.Do().Error(); err != nil {
+		return nil, err
+	}
+
+	return list, nil
+}
+
+func (g *GuideService) getPipelineYml(app *apistructs.ApplicationDTO, userID string, branch string, findPath string) ([]PipelineYml, error) {
+	var path string
+	if findPath == "" {
+		path = fmt.Sprintf("/wb/%v/%v/tree/%v", app.ProjectName, app.Name, branch)
+	} else {
+		path = fmt.Sprintf("/wb/%v/%v/tree/%v/%v", app.ProjectName, app.Name, branch, findPath)
+	}
+
+	treeData, err := g.bdl.GetGittarTreeNode(path, strconv.Itoa(int(app.OrgID)), true, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []PipelineYml
+	for _, entry := range treeData.Entries {
+		if !strings.HasSuffix(entry.Name, ".yml") {
+			continue
+		}
+		if findPath == "" && entry.Name != apistructs.DefaultPipelineYmlName {
+			continue
+		}
+		list = append(list, PipelineYml{
+			YmlName: entry.Name,
+			YmlPath: findPath,
+		})
+	}
+	return list, nil
+}
+
+func (g *GuideService) checkBranchRule(branch string, projectID int64) (bool, error) {
+	branchRules, err := g.branchRuleSve.Query(apistructs.ProjectScope, int64(projectID))
+	if err != nil {
+		return false, err
+	}
+	_, err = diceworkspace.GetByGitReference(branch, branchRules)
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (g *GuideService) checkPipelineYml(app *apistructs.ApplicationDTO, branch, userID string) (bool, error) {
+	ymls, err := g.ListPipelineYml(app, branch, userID)
+	if err != nil {
+		return false, err
+	}
+	if len(ymls) == 0 {
+		return false, fmt.Errorf("the pipeline yml is not exists")
```

Comment:
I guess `return false, nil` is more proper, because you check `ok == false` later

Reply:
done